addons/website/tests/test_crawl.py

```diff
@@ -53,8 +53,9 @@ def crawl(self, url, seen=None, msg=''):
         if seen is None:
             seen = set()
 
-        url_slug = re.sub(r"[/](([^/=?&]+-)?[0-9]+)([/]|$)", '/<slug>/', url)
-        url_slug = re.sub(r"([^/=?&]+)=[^/=?&]+", '\g<1>=param', url_slug)
+        url_slug = re.sub(r"[/](([^/=?&]+-)?[0-9]+)([/]?)", '/<slug>/', url)
```

Comment:
Shouldn't it be
```suggestion
        url_slug = re.sub(r"[/](([^/=?&]+-)?[0-9]+)([/]|$|\?|#)", '/<slug>/', url)
```
So it matches `word-id` slug only and not `word-id-word`?
![image](https://github.com/odoo/odoo/assets/30048408/46d906a5-88d4-4c90-8d87-3263c779837a)

Instead of the current diff (see second line):
![image](https://github.com/odoo/odoo/assets/30048408/61019f1a-b620-410a-a7fd-5dcd780a9a7c)

For tracking purpose, code before this PR, see 2 last occurrences, this is the issue fixed here:
![image](https://github.com/odoo/odoo/assets/30048408/50730f11-f7d1-4320-9a3d-4aa7d85eeb21)

Reply:
Merged you regexp with the \3 in sub to keep the correct char in /slug[...] and added a rstrip on it to consider url with trailing # ? / as identical to url without it.


```
https://www.odoo.com/forum/aide-1/post/post-2
-> https://www.odoo.com/forum/<slug>/post/<slug>

https://www.odoo.com/forum/aide-1/post/post-2?
-> https://www.odoo.com/forum/<slug>/post/<slug>

https://www.odoo.com/forum/aide-1/post/post-2?a=b&c=d
-> https://www.odoo.com/forum/<slug>/post/<slug>?a=param&c=param

https://www.odoo.com/forum/aide-1/post/post-2#
-> https://www.odoo.com/forum/<slug>/post/<slug>

https://www.odoo.com/forum/aide-1/post/post-2#a=b&c=d
-> https://www.odoo.com/forum/<slug>/post/<slug>#a=param&c=param

https://www.odoo.com/page/aide-1-bla
-> https://www.odoo.com/page/aide-1-bla

https://www.odoo.com/page/aide-1-bla?a=b
-> https://www.odoo.com/page/aide-1-bla?a=param
```